src/libraries/System.Globalization.Extensions/tests/IdnMapping/Data/Unicode_13_0/Unicode_13_0_IdnaTest.cs

```diff
@@ -0,0 +1,58 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    /// <summary>
+    /// Class to read data obtained from http://www.unicode.org/Public/idna.  For more information read the information
+    /// contained in Data\Unicode_13_0\IdnaTest_13.txt
+    ///
+    /// The structure of the data set is a semicolon delimited list with the following columns:
+    ///
+    /// Column 1: source -          The source string to be tested
+    /// Column 2: toUnicode -       The result of applying toUnicode to the source,
+    ///                             with Transitional_Processing=false.
+    ///                             A blank value means the same as the source value.
+    /// Column 3: toUnicodeStatus - A set of status codes, each corresponding to a particular test.
+    ///                             A blank value means [] (no errors).
+    /// Column 4: toAsciiN -        The result of applying toASCII to the source,
+    ///                             with Transitional_Processing=false.
+    ///                             A blank value means the same as the toUnicode value.
+    /// Column 5: toAsciiNStatus -  A set of status codes, each corresponding to a particular test.
+    ///                             A blank value means the same as the toUnicodeStatus value.
+    ///                             An explicit [] means no errors.
+    /// Column 6: toAsciiT -        The result of applying toASCII to the source,
+    ///                             with Transitional_Processing=true.
+    ///                             A blank value means the same as the toAsciiN value.
+    /// Column 7: toAsciiTStatus -  A set of status codes, each corresponding to a particular test.
+    ///                             A blank value means the same as the toAsciiNStatus value.
+    ///                             An explicit [] means no errors.
+    ///
+    /// If the value of toUnicode or toAsciiN is the same as source, the column will be blank.
+    /// </summary>
+    public class Unicode_13_0_IdnaTest : Unicode_IdnaTest
+    {
+        public Unicode_13_0_IdnaTest(string line, int lineNumber)
+        {
+            var split = line.Split(';');
+
+            Type = PlatformDetection.IsNlsGlobalization ? IdnType.Transitional : IdnType.Nontransitional;
+
+            Source = EscapedToLiteralString(split[0], lineNumber);
+
+            UnicodeResult = new ConformanceIdnaUnicodeTestResult(EscapedToLiteralString(split[1], lineNumber), Source, EscapedToLiteralString(split[2], lineNumber), string.Empty);
+            ASCIIResult = new ConformanceIdnaTestResult(EscapedToLiteralString(split[3], lineNumber), UnicodeResult.Value, EscapedToLiteralString(split[4], lineNumber), UnicodeResult.StatusValue);
+
+            if (Type == IdnType.Transitional)
```

Comment:
>Transitional [](start = 32, length = 12)

I am confused here. 13.0 is going to be used on ICU which we'll enable non transitional flag. why we are testing for transitional here?

Reply:
Oh this was to keep it consistent in case we moved to a newer version of Windows which uses Unicode_13. That way the NLS tests would use Unicode_13.

Reply:
could you add "Windows" to the class name then? I would say if we are not going to use this now, then remove it till we need it.

---
In reply to: [435561007](https://github.com/dotnet/runtime/pull/37281#discussion_r435561007) [](ancestors = 435561007)

Reply:
I was trying to avoid cases like Unicode_11 where we accidentally loose test data because of the wrong IdnType.

> could you add "Windows" to the class name then?

So you mean adding a new Unicode_13_0_IdnaTest? 

I think it makes sense to instead add a comment explaining that Nls uses Transitional.

Reply:
I was hoping to have classes specific for Windows(NLS) and classes specific for ICU. this separation can show the difference without need to understand what is going on. but anyway, I don't think we need to invest more in these tests if we are going to change it.

---
In reply to: [435564563](https://github.com/dotnet/runtime/pull/37281#discussion_r435564563) [](ancestors = 435564563)

Reply:
>  I don't think we need to invest more in these tests if we are going to change it.

Yeah, I agree. Adding separation of classes would make sense if we wanted to stick with this model of testing in the future, but it is a big change, so I will either leave it as a subsequent PR in the case we wanted to keep these tests